delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/ResourceSourceParser.scala

```diff
@@ -16,6 +17,53 @@ import monix.bio.{IO, UIO}
 
 trait ResourceSourceParser {
 
+  /**
+    * Computes the id from the segment or the payloadId
+    * If both are provided, we make sure, they don't clash
+    *
+    * @param idSegment        the id provided in the segment
+    * @param project          the project
+    * @param idPayload        the id provided by the payload
+    */
+  def computeId[R](idSegment: Option[IdSegment], project: Project, idPayload: Option[Iri])(implicit
+      uuidF: UUIDF,
+      rejectionMapper: Mapper[ResourceRejection, R]
+  ): IO[R, Iri] = {
+    val result = idSegment match {
+      case None          => getOrGenerateId(idPayload, project)
+      case Some(segment) =>
+        IO.fromOption(
+          segment.toIri(project.apiMappings, project.base),
+          InvalidResourceId(segment.asString)
+        ).flatMap { iri =>
+          idPayload match {
+            case Some(p) if iri != p =>
+              IO.raiseError(UnexpectedResourceId(iri, p))
+            case _                   =>
+              IO.pure(iri)
+          }
+        }
+    }
+
+    result.leftMap(rejectionMapper.to)
+  }
+
+  def expandIri[R](segment: IdSegment, project: Project)(implicit
+      rejectionMapper: Mapper[ResourceRejection, R]
+  ): IO[R, Iri] =
+    IO.fromOption(
+      segment.toIri(project.apiMappings, project.base),
+      rejectionMapper.to(InvalidResourceId(segment.asString))
```

Comment:
Why this is fixed to `InvalidResourceId`. In this case the wrapping thing does not make sense.
If a resolver has the same Id in the payload as in the segment, I would call it a `InvalidResolverId`. Same for Schemas.
It makes sense to wrap the other rejections because it is the responsibility of the other API bundles (project and organizations).

Reply:
They are using the same piece of code which could be considered like a common submodule, I don't see the need to define different rejections for every type of resource that is using it.

Reply:
I don't think it hurts duplication here. And even more in this case, it is not a resource, it is a resolver and the rejection type should be `InvalidResolverId` and the message too should say stuff about the resolver, not the resource